tests/data/version_condition2.json

```diff
@@ -0,0 +1,13 @@
+[
+  {
+    "condition": ">=",
+    "release": "1",
+    "version": "0.0"
+  },
+  {
+    "package_name": "apparmor-parser",
+    "condition": ">=",
+    "version": "2.12.2"
+  }
+
+]
```

Comment:
There should be an error condition test for
```
{
    "condition": ">=",
    "release": "1",
    "version": "0.0",
    "package_name": "apparmor-parser",
    "condition": ">=",
    "version": "2.12.2"
}
```
This will also defeat the idea of list conversion ;) as one item in the list will still contain two conditions that I don't think we can process.

Also we should have a test that catches a syntax error in the json file.

Reply:
Included the proper error handling to validate the accepted json inputs (just the type, that has to be a list now) and fail in any unexpected case. I removed also the list conversion if you provide a json object, now the script only accepts lists.

Not sure I follow you completely with this tests with repeated keys in the json object, we can detect the repeated keys if we want to (with a hook in json.load/json.loads), but the default behavior would be that the values for the repeated keys get overwritten the second time the key appears.
Do we want to raise an exception in such cases or validate the provided json vs an schema?